### PR TITLE
Flexible fields for roiInfo

### PR DESCRIPTION
**Pull request by romainFr**
Incorporate upstream/downstream as needed (depending on their presence on the server) in neuprint_get_meta, fixes tests.
This rests on a couple of utility functions, neuprint_get_fields that checks for the presence of fields in neuron objects, and dfFields which translate them into our usual column names.

### CODE REVIEW

**Comment by jefferis** [comment]
Thanks for this @romainFr. I can’t review now, but did you memoise the check for which fields must be used?

**Comment by romainFr** [comment]
I didn't memoise, but the function that pulls the fields is not called inside repeated calls (like extract_connectivity_df) and it's pretty fast. Let me know if you still think memoise would help.

**Comment by jefferis** [comment]
Hi @romainFr, thanks a lot for this! I'm still reviewing, but I did think it was worth memoising.

File: tests/testthat/test-name.R [hunk 1]
```diff
@@ -51,5 +42,5 @@ test_that("test name searches ", {
 })
 
 test_that("test bad dataset specification ", {
-  expect_warning(neuprint_search(".*DA2.*", dataset = 'hemibrain'))
+  expect_warning(neuprint_search(".*DA2.*", dataset = 'hemibrain1'))
 })
```
Comment (jefferis) [inline tests/testthat/test-name.R]: @romainFr is this a valid dataset on the test server out of curiosity?

Comment (romainFr) [reply tests/testthat/test-name.R]: Yes, that's the "work in progress" server.

File: R/connectivity.R [hunk 2]
```diff
@@ -472,16 +472,16 @@ neuprint_get_shortest_paths <- function(body_pre,body_post,weightT=5,roi=NULL,da
...
     return(NULL)
   }
   rois <- unique(rois) #this takes care if both the input and output ROIs are same..
   a <- unlist(jsonlite::fromJSON(json))
-  roicols <- c(t(outer(rois, c("pre", "post","upstream","downstream"), paste, sep=".")))
+  roicols <- c(t(outer(rois,postFix, paste, sep=".")))
   values <- tibble::as_tibble(as.list(structure(rep(0, length(roicols)), .Names=roicols)))
   for(roi in rois){
-    thisroicols <- paste0(roi,c(".pre",".post",".upstream",".downstream"))
+    thisroicols <- paste0(roi,".",postFix)
```
Comment (jefferis) [inline R/connectivity.R]: looks like thisroicols never got used (I don't think that was your fault @romainFr!)

Comment (romainFr) [reply R/connectivity.R]: Oh true. Must be a remnant of a previous implementation. But wait, didn't you change it back in https://github.com/natverse/neuprintr/pull/95/commits/ecb52ac7fb5572f5dd7da5e28ce9ea7fa4bb0ebf ?

File: tests/testthat/test-name.R (hunk 1 above)
Comment (jefferis) [reply tests/testthat/test-name.R]: Sorry I meant is `hemibrain` a valid dataset on the work in progress server?

**Pull request merged**